### DISCUSSION TITLE
Workflow Testing Elsa V3

### DISCUSSION THREAD

**Discussion by PontiusTheBarbarian** [elsa-workflows/elsa-documentation#70] [Q&A]
Hi, 

I am in search of the equivalent packages for workflow testing/visualisation you had in Elsa V2 - `Elsa.WorkflowTesting.Api` and `services.AddWorkflowTestingServices();` -  but for V3? Are these available or are they on your radar for a release anytime soon?

Many thanks

Liam.

**Comment by sfmskywalker** [comment]
If I remember correctly ,those packages are used to enable running a workflow straight from the designer (please do correct me if I am wrong).

If that's correct, then I have good news: for V3, you don't need separate packages for that; you can run workflows from the designer right out of the box 😄

**Comment by PontiusTheBarbarian** [comment]
Hi @sfmskywalker,

Thank you for your reply, you are probably right. I havent actually used this in older versions I noticed that on [this thread](https://github.com/elsa-workflows/elsa-core/issues/671#issuecomment-926558052) that someone referred to a testing tab and the ability execute from the designer. I guess I was looking for something that similarly referred to 'Test' from within the designer but what you've said also makes sense.

On that note, in the limitations on your V3 site you mention : 'Starting workflows from the designer is currently supported only for workflows that do not require input and do not start with a trigger; this is planned for a future release.', do you have any idea when this will be released and/or a workaround for testing workflows with inputs?

Many thanks, Liam
